apiserver/facades/client/client/status.go

```diff
@@ -1069,14 +1078,24 @@ func (context *statusContext) processUnits(units map[string]*state.Unit, applica
 
 func (context *statusContext) processUnit(unit *state.Unit, applicationCharm string) params.UnitStatus {
 	var result params.UnitStatus
-	addr, err := unit.PublicAddress()
-	if err != nil {
-		// Usually this indicates that no addresses have been set on the
-		// machine yet.
-		addr = network.Address{}
-		logger.Debugf("error fetching public address: %v", err)
+	if unit.ShouldBeAssigned() {
+		addr, err := unit.PublicAddress()
+		if err != nil {
+			// Usually this indicates that no addresses have been set on the
+			// machine yet.
+			addr = network.Address{}
+			logger.Debugf("error fetching public address: %v", err)
+		}
+		result.PublicAddress = addr.Value
```

Comment:
So, if err != nil, we'll fall through to here and result.PublicAddress will be ? ""?